ramips: add support for Phicomm K2 new flash layout

DragonBluep opened the pull request:
Descriptions：
Phicomm K2 (PSG1218) got a new "permanent_config" partition after update
firmware to v22.5. This partition located in front of the firmware partition,
same as The Phicomm K2P and K2G. Due to this change the new bootloader can't
load previous firmware any more. This commit is aimed at add support for
Phicomm K2 which official firmware version is 22.5.x or newer. For which runs
old firmware version, just update OpenWrt that has a prefix of "k2-v22.4".
For uniform naming, this commit also changed the model name PSG1218 to a more
recognizable name K2, refer to Phicomm K2G, K2P K2T.

OpenWrt selection table:
official firmware version           OpenWrt
v21.4.x.x or older              phicomm_k2-v22.4
v22.5.x.x or newer              phicomm_k2-v22.5

Installation:
Same as Phicomm K2G, K2P, PSG1208.
a. TFTP + U-Boot
b. Open telnet by some web page vulnerability (Search Baidu by key words "K2 telnet"),
and then we can upload firmware image to /tmp and write it to firmware partition with
mtd instruction.

adschm commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -802,6 +802,16 @@ define Device/ohyeah_oy-0001
+  IMAGE_SIZE := 7552k
+  DEVICE_VENDOR := Phicomm
+  DEVICE_MODEL := K2
+  DEVICE_VARIANT:= v22.5+
```
What's that?

DragonBluep replied:
Before official firmware version v22.5, bootloader load kernel from 0x50000. But new version Phicomm K2 has an additional partition named "permanent_config", and bootloader is not compatible with PSG1218 anymore. I don't have a perfect idea to distinguish this difference so I add a DEVICE_VARIANT. But if we can add some additional illustration on openwrt wiki, I think this one can be removed.

adschm replied:
So, we need a phicomm_k2-v22.5-plus and a phicomm_k2-v22.4-minus then?

DragonBluep replied:
Looks a little strange. I found there are only four main software versions, v21.4.x.x, v22.4.x.x, v22.5.x.x and v22.6.x.x. 
And this devices was end of support by official site for a long time. Which of the following two options do you think is more appropriate? I can't decide.
phicomm_k2-v22.5-v22.6 & phicomm_k2-v21.4-v22.4
phicomm_k2-v22.5-plus & phicomm_k2-v22.4-minus

DragonBluep replied:
Should I rename psg1218 to phicomm_k2-v22.4-minus now?

adschm replied:
My point was mainly that we need to find a "good" naming scheme here. I'm not sure what would be "good" in this case yet, though.

DragonBluep replied:
It seems there are only 5 main software versions, v21.4.x.x, v22.3.x.x, v22.4.x.x, v22.5.x.x and v22.6.x.x. And this devices was end of support by official site for a long time. I think phicomm_k2-v22.5-v22.6 & phicomm_k2-v21.4-v22.4 can be a better choice.

DragonBluep commented:
"phicomm_k2-v22.5" and "phicomm_k2-v22.4" prefix are explicit enough to help user to choose suitable firmware by compare their official firmware's version, and these prefix are less verbose. It's so obvious and I believe no one will install "phicomm_k2-v22.4.xxx.bin" on v22.5.x.x or newer version firmware after comparing the two prefixes.

DragonBluep committed:
ramips: add new flash layout support to Phicomm K2

Descriptions：
Phicomm K2 (PSG1218) got a new "permanent_config" partition after update
firmware to v22.5. This partition located in front of the firmware partition,
same as The Phicomm K2P and K2G. Due to this change the new bootloader can't
load previous firmware any more. This commit is aimed at add support for
Phicomm K2 which official firmware version is 22.5.x or newer. For which runs
old firmware version, just update OpenWrt that has a prefix of "k2-v22.4".
For uniform naming, this commit also changed the model name PSG1218 to a more
recognizable name K2, refer to Phicomm K2G, K2P K2T.

OpenWrt selection table:
official firmware version           OpenWrt
v21.4.x.x or older              phicomm_k2-v22.4
v22.5.x.x or newer              phicomm_k2-v22.5

Installation:
Same as Phicomm K2G, K2P, PSG1208.
a. TFTP + U-Boot
b. Open telnet by some web page vulnerability (Search Baidu by key words "K2 telnet"),
and then we can upload firmware image to /tmp and write it to firmware partition with
mtd instruction.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

adschm commented:
Thanks, I will merge it with minor adjustments today.
